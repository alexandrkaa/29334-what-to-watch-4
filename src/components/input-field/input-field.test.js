import React from 'react';
import renderer from 'react-test-renderer';
import InputField from './input-field.jsx';


it(`<InputField /> should match snapshot`, () => {
  const mockData = {
    id: `login`,
    label: `User login`,
    type: `text`,
    placeholder: `Type login, please`,
    value: ``,
    onChange: jest.fn(),
    isDisabled: false,
  };
  const {id, label, type, placeholder, value, onChange, isDisabled} = mockData;
  const tree = renderer
    .create(
        <InputField
          id={id}
          label={label}
          type={type}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          isDisabled={isDisabled}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
