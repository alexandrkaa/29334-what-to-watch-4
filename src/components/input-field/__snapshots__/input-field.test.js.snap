// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> should match snapshot 1`] = `
<div
  className="sign-in__field"
>
  <input
    className="sign-in__input"
    disabled={false}
    id="login"
    name="login"
    onChange={[Function]}
    placeholder="Type login, please"
    type="text"
    value=""
  />
  <label
    className="sign-in__label visually-hidden"
    htmlFor="login"
  >
    User login
  </label>
</div>
`;
